libs/wingsdk/src/util/util.ts

```diff
@@ -1,9 +1,161 @@
+import { exec, execFile, spawn } from "child_process";
 import { createHash } from "crypto";
+import { promisify } from "util";
 import { nanoid, customAlphabet } from "nanoid";
 import { v4 } from "uuid";
 import { InflightClient } from "../core";
 import { Duration, IInflight } from "../std";
 
+const execPromise = promisify(exec);
+const execFilePromise = promisify(execFile);
+
+/**
+ * Describes what to do with a standard I/O stream for a child process.
+ */
+export enum Stdio {
+  /**
+   * The child inherits from the corresponding parent descriptor.
+   */
+  INHERIT = "inherit",
+  /**
+   * A new pipe should be arranged to connect the parent and child processes.
+   */
+  PIPED = "pipe",
+  /**
+   * This stream will be ignored. This is the equivalent of attaching the stream to /dev/null.
+   */
+  NULL = "ignore",
+}
+
+/**
+ * Base command options.
+ */
+export interface CommandOptions {
```

Comment:
Add an option to control whether a non zero exit code results in an exception or not

Reply:
I think this should apply only to `ShellOptions`, since both `exec()` and `spawn()` return an `Output` struct which contains the error code inside, so it would be up to the user whether to throw or not based on the returned `Output.status`.